gap/projective/almostsimple.gi

```diff
@@ -1429,3 +1429,126 @@ FindHomMethodsProjective.SporadicsByOrders := function(ri,G)
   od;
   return false; # FIXME: false = NeverApplicable here really correct?
 end;
+
+#! @BeginChunk NameSporadic
+#! This method returns a list of sporadic simple groups that <A>ri</A>
+#! could be. It does not recognise the monster and the baby monster group. It
+#! is based on the magma v2.24.10 function <C>RecognizeSporadic</C>.
+FindHomMethodsProjective.NameSporadic := function(ri)
+    local orders, setOfOrders, maximalOrders, counter, namesOfPossibleSporadics, i, j;
+    orders := [];
+    #do we need SetPseudoRandomStamp?
+    for i in [1 .. 500] do
+        Add(orders, RandomElmOrd(ri, "NameSporadic", false).order);
+    od;
+    # Compute maximal orders. Maximal in the sense that it does not divide the
+    # order of another group element.
+    setOfOrders := Immutable(Set(orders));
+    maximalOrders := [];
+    for i in setOfOrders do
+        counter := 0;
+        for j in setOfOrders do
+            if j mod i = 0 then
+                counter := counter + 1;
+            fi;
+        od;
+        if counter = 1 then
+            Add(maximalOrders, i);
+        fi;
+    od;
```

Comment:
This loops is somewhat inefficient, although it might not matter if `setOfOrders` is always small: if there are N orders, we perform N^2 divisions and checks. But clearly N over 2 suffice, so about a factor 2 in performance could be gained. Moreover, one can do away with counter. Something like this might work (untested):
```suggestion
    for i in [1..Length(setOfOrders)] do
        isMaximal := true;
        for j in [i+1..Length(setOfOrders)] do
            if j mod i = 0 then
                isMaximal := false;
                break;
            fi;
        od;
        if isMaximal then
            Add(maximalOrders, i);
        fi;
    od;
```

Reply:
I'm adding this. j and i have to be setOfOrders[j] or ..[i] though.